stdlib/bytes.ml

```diff
@@ -97,19 +97,30 @@ let iter f a =
 let iteri f a =
   for i = 0 to length a - 1 do f i (unsafe_get a i) done
 
+let () = assert (3 * Sys.max_string_length <= max_int)
```

Comment:
I think the idea is that the problem is not in this code, but in the choice of setting `Sys.max_string_length` to `max_int`, i.e. BucleScript should set `max_string_length` to `max_int/3`.